### ISSUE TITLE
🐞multi-file plan triggers incorrect input validation errors

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#2386]
## Problem

When a plan is split across multiple files, it fails incorrectly with input validation errors ("xxx not set"). The same plan, when combined into a single file, works fine.

### Dagger version

```
$ dagger version
dagger devel (ff0beef6) linux/amd64
```

### What I did

1. Setup a new project to reproduce

```
cd $(mktemp -d)
dagger project init
```

2. Add repro config

```
cat > foo.cue <<-EOF
package main

import (
        "dagger.io/dagger"
        "dagger.io/dagger/core"
)

dagger.#Plan & {
        client: filesystem: ".": read: contents: dagger.#FS
        actions: build: core.#Nop & {
                input: client.filesystem.".".read.contents
        }
        actions: deploy: core.#Nop & {
                input: actions.build.output
        }
}
EOF
```

```
cat > bar.cue <<-EOF
package main

import (
        "dagger.io/dagger"
        "dagger.io/dagger/core"
)

dagger.#Plan & {
        actions: deploy: core.#Nop & {
                input: actions.build.output
        }
}
EOF
```

3. Install deps

```
dagger project update
```

4. Run deploy action

```
dagger do deploy
```

### What I expected to see

Expected output for step 4:

```
$ dagger do deploy
[✔] client.filesystem.".".read                                                                                                                    0.0s
[✔] actions.build                                                                                                                                 0.0s
[✔] actions.deploy                          
```

### What I see instead

Actual output of step 4:

```
$ dagger do deploy
failed to load plan: "client.filesystem.\".\".read" is not set


Usage: 
  dagger do [flags]


Flags:
      --cache-from stringArray   External cache sources (eg. user/app:cache, type=local,src=path/to/dir)
      --cache-to stringArray     Cache export destinations (eg. user/app:cache, type=local,dest=path/to/dir)
  -h, --help                     help for do
      --no-cache                 Disable caching
  -p, --plan string              Path to plan (defaults to current directory) (default ".")
      --platform string          Set target build platform (requires experimental)
  -w, --with stringArray
Global Flags:
      --experimental        Enable experimental features
      --log-format string   Log format (auto, plain, tty, json) (default "auto")
  -l, --log-level string    Log level (default "info")
```

**Comment by aluzzardi** [comment]
Root cause is #2263 -- multiple `#Plan` in the same project breaks CUE /cc @helderco 

Removing `dagger.#Plan &` from `bar.cue` fixes the issue

**Comment by shykes** [comment]
> Root cause is #2263 -- multiple `#Plan` in the same project breaks CUE /cc @helderco
> 
> Removing `dagger.#Plan &` from `bar.cue` fixes the issue

Thanks! That will probably unblock me :)

We should still consider this a bug, right?

**Comment by aluzzardi** [comment]
> We should still consider this a bug, right?

Yes. However I believe the bug is already tracked by #2263

**Comment by shykes** [comment]
Roger that.
